### PR TITLE
Added popular room recommendations

### PR DESCRIPTION
**Pull request by louissullivan4**
Added the following:
> Class that randomly generates rooms and accounts
> Changed endpoint to addAccountToRoom from needing account object to an account id parameter
> Added the popularity reccomendations which calls the backend to return all the rooms in the database in descending order where the order is based on the number of accounts related to that room i.e. highest number of accounts returned first. It will then add the 5 most popular rooms to the person's reccomendations.

### CODE REVIEW

File: demo/demo/src/main/java/com/example/demo/GenerateRandomData.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        }
+        return sb.toString();
+    }
+
+    private static <T> T getRandomElement(T[] array) {
+        Random random = new Random();
+        int index = random.nextInt(array.length);
+        return array[index];
+    }
+}
```
Comment (CianMcDonald) [inline demo/demo/src/main/java/com/example/demo/GenerateRandomData.java:84]: That must have taken a while to make lol

File: demo/demo/src/main/java/com/example/demo/controller/AccountController.java [hunk 2]
```diff
@@ -70,10 +63,10 @@ ResponseEntity<List<Room>> getAllRoomsByAccountId(@PathVariable Long id) {
         return new ResponseEntity<>(rooms, HttpStatus.OK);
     }
 
-    @PostMapping("/rooms/{id}/accounts")
-    ResponseEntity<Account> addAccountToRoom(@PathVariable Long id, @RequestBody Account accountRequest) {
+    @PostMapping("/rooms/{id}/accounts/{accountId}")
+    ResponseEntity<Account> addAccountToRoom(@PathVariable Long id, @PathVariable Long accountId) {
         Account account = roomRepository.findById(id).map(room -> {
-           long accountId = accountRequest.getId();
+//           long accountId = accountRequest.getId();
```
Comment (CianMcDonald) [inline demo/demo/src/main/java/com/example/demo/controller/AccountController.java:69]: Perf just removed commented bits tho

File: demo/demo/src/main/java/com/example/demo/repository/RoomRepository.java [hunk 3]
```diff
@@ -7,4 +7,6 @@
 
 public interface RoomRepository extends JpaRepository<Room, Long> {
     List<Room> findRoomsByAccountsId(Long accountId);
+
+    List<Room> findAllByOrderByAccountsDesc();
```
Comment (CianMcDonald) [inline demo/demo/src/main/java/com/example/demo/repository/RoomRepository.java:11]: Haha what a funny way they make ya write that.

**Review by CianMcDonald — APPROVED** [review]
Very good Louis sir

File: demo/demo/src/main/java/com/example/demo/repository/RoomRepository.java (hunk 3 above)
Comment (louissullivan4) [reply demo/demo/src/main/java/com/example/demo/repository/RoomRepository.java:11]: so painful this took ages to figure out

File: demo/demo/src/main/java/com/example/demo/GenerateRandomData.java (hunk 1 above)
Comment (louissullivan4) [reply demo/demo/src/main/java/com/example/demo/GenerateRandomData.java:84]: lots of googling my errors

**Pull request merged**